app/client/src/sagas/WidgetBlueprintSagas.ts

```
@@ -139,13 +165,44 @@ export function* executeWidgetBlueprintOperations(
+  operation: BlueprintOperation,
+) {
+  switch (operation.actionType) {
+    case BlueprintOperationActionTypes.CREATE_OR_UPDATE_DATASOURCE_WITH_ACTION:
+      if (
+        !operation.payload?.pluginPackageName ||
+        !operation.payload?.actionConfig
+      )
+        return;
+
+      const { actionConfig, datasourceName, pluginPackageName } =
+        operation.payload;
```

Biome

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)